xarray/core/dataset.py

```diff
@@ -5922,6 +5983,56 @@ def reduce(
         reduced : Dataset
             Dataset with this object's DataArrays replaced with new DataArrays
             of summarized data and the indicated dimension(s) removed.
+
+        Examples
+        --------
+        >>> dataset = xr.Dataset(
+        ...     {
+        ...         "math_scores": (
+        ...             ["student", "test"],
+        ...             [[90, 85, 92], [78, 80, 85], [95, 92, 98]],
+        ...         ),
+        ...         "english_scores": (
+        ...             ["student", "test"],
+        ...             [[88, 90, 92], [75, 82, 79], [93, 96, 91]],
+        ...         ),
+        ...     },
+        ...     coords={
+        ...         "student": ["Alice", "Bob", "Charlie"],
+        ...         "test": ["Test 1", "Test 2", "Test 3"],
+        ...     },
+        ... )
+
+        # Calculate the 75th percentile of math scores for each student using np.percentile
+        >>> percentile_scores = dataset.reduce(np.percentile, q=75, dim="test")
+        >>> percentile_scores
+        <xarray.Dataset>
+        Dimensions:         (student: 3)
+        Coordinates:
+          * student         (student) <U7 'Alice' 'Bob' 'Charlie'
+        Data variables:
+            math_scores     (student) float64 91.0 82.5 96.5
+            english_scores  (student) float64 91.0 80.5 94.5
+
+Calculating skewness of math scores for each student using the ``scipy.stats.skew`` function using the above mentioned dataset
+
+        # To use the `skew` function, you need to import it from the `scipy.stats` module
+        >>> from scipy.stats import skew
+
+        # Combine the scores of both subjects into a single variable
+        >>> combined_scores = xr.concat([dataset["math_scores"], dataset["english_scores"]], dim="subject")
+
+        # Calculate the skewness of scores for all students
+        >>> skewness_scores = combined_scores.reduce(skew, dim=("test", "student"))
```

Comment:
We should be able to do this without `concat`, using `Dataset.reduce`. But when we tried this, we ran into an error. (See issue https://github.com/pydata/xarray/issues/7954)

For now @harshitha1201 let's remove this example, and come back to it later in another Pull Request, so that we can get the other stuff merged.